Tests/Mocks/MockCardNumberViewModel.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  MockCardNumberViewModel.swift
+//  FramesTests
+//
+//  Created by Harry Brown on 13/07/2022.
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+@testable import Frames
+
+final class MockCardNumberViewModel: CardNumberViewModelProtocol {
+  private(set) var textFieldUpdateCalledWith: String?
```

Comment:
Not important right now, but if we use arrays for these `calledWith` records, we can then make sure we also count the number of calls received, as well as check progress instead of just last value received.